pkg/trace/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/datadog-agent/pkg/trace
 
-go 1.20
+go 1.21
```

Comment:
This (and other updates of the `go` directive in Go modules used by the Datadog exporter) will break all users of the OpenTelemetry Collector Builder that use the Datadog exporter and use Go 1.20. Collector components officially support all supported Go versions (1.20 and 1.21) as stated [here](https://github.com/open-telemetry/opentelemetry-collector?tab=readme-ov-file#compatibility).

We can just update the version used for building, but keep the modules compatible with Go 1.20 until support is dropped for it. This is what we do in the OpenTelemetry Collector: use the latest Go version for building the binaries, but keep compatibility with Go 1.20 at the library level

Reply:
Following our conversation, we'll keep the go.mod's go directive of modules used by the exporter to 1.20, and we'll wait until the CI checks whether those modules can build with 1.20 before merging.
Note that we have a hard deadline to merge this before the release freeze.